File: src/gen-lexeduct-transformers.js
#!/usr/bin/env node

fs = require('fs');

function write(s) {
    process.stdout.write(s + "\n");
}

write("module = {};");
write("transformer = {};");
var dirname = 'transformers';
var files = fs.readdirSync(dirname);
for (var i = 0; i < files.length; i++) {
    var filename = files[i];
    var transformerName = filename.split('.js')[0];
    var text = fs.readFileSync(dirname + '/' + filename);
    write(text);
    write("transformer['" + transformerName + "'] = module.exports;");
}

File: src/gen-lexeduct-transformers.py
#!/usr/bin/env python3

import os
import sys


def write(s):
    sys.stdout.write(s + "\n")


def main(args):
    write("module = {};")
    write("transformer = {};")
    dirname = "transformers"
    for root, dirnames, filenames in os.walk(dirname):
        dirnames[:] = []
        filenames.sort()
        for filename in filenames:
            transformer_name = filename.split('.js')[0]
            with open(os.path.join(dirname, filename)) as f:
                write(f.read())
            write("transformer['{}'] = module.exports;".format(transformer_name))


if __name__ == '__main__':
    main(sys.argv[1:])
